source/extensions/listener_managers/listener_manager/active_stream_listener_base.cc

```diff
@@ -112,6 +112,8 @@ void ActiveTcpConnection::onEvent(Network::ConnectionEvent event) {
   // Any event leads to destruction of the connection.
   if (event == Network::ConnectionEvent::LocalClose ||
       event == Network::ConnectionEvent::RemoteClose) {
+    connection_->connectionInfoSetter().setDownstreamTransportFailureReason(
```

Comment:
Here is possible that the connection is closed successfully without `transportFailure`, right? Do we want to treat that differently? i.e., not reporting error.

We can check` if (!connection_->transportFailureReason.empty())` first

Reply:
Yea , I missed that you already check !empty in substitution_formatter.cc. Check here is not necessary